commons/src/lib.rs

```diff
@@ -40,14 +49,31 @@ pub fn ensure_query_params(params: &HashSet<String>, query: &str) -> Fallible<()
         .collect();
 
     // Make sure all mandatory parameters are present.
-    ensure!(
-        params.is_subset(&query_keys),
-        "mandatory parameters missing"
-    );
+    if !required_params.is_subset(&query_keys) {
+        return Err(GraphError::MissingParams);
+    }
 
     Ok(())
 }
 
+/// Make sure client requested the relevant content type.
+pub fn ensure_content_type(
+    headers: &actix_web::http::HeaderMap,
+    content_type: &'static str,
+) -> Result<(), GraphError> {
+    let content_json = header::HeaderValue::from_static(content_type);
+
+    if !headers
+        .get(header::ACCEPT)
+        .map(|accept| accept == content_json)
```

Comment:
opportunity: pull this into a declared table/struct of the accepted content types

Reply:
I started plumbing through an `HashSet` for this, but I'm not totally satisfied with that.
Sticking close to the current logic, we would be constructing that single-member set on each request (without any actual need to).
We could either move that into AppState, or tackle this later on only if we actually end up with a multi-content endpoint.

Any further feedback/preference from your side?

Reply:
Let's stick to the current approach since we have no plans to handle different content types right now.

Reply:
Ack.